File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot/IntakeSubsystems/KickerSubsystem.java
package org.firstinspires.ftc.teamcode.Robot.IntakeSubsystems;

import dev.nextftc.core.commands.Command;
import dev.nextftc.core.subsystems.Subsystem;
import dev.nextftc.hardware.impl.ServoEx;
import dev.nextftc.hardware.positionable.SetPosition;

public class KickerSubsystem implements Subsystem {
    public static KickerSubsystem INSTANCE = new KickerSubsystem();
    public KickerSubsystem() { }

    public ServoEx kicker = new ServoEx("k");

    public static double DOWN_POS = 0;
    public static double UP_POS = 0.27;

    public Command kickBall = new SetPosition(kicker, UP_POS).requires(this);
    public Command reset = new SetPosition(kicker, DOWN_POS).requires(this);

    @Override
    public void initialize() {
        kicker.setPosition(0);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Systems/ShooterSubsystems/KickerSubsystem.kt
package org.firstinspires.ftc.teamcode.Systems.ShooterSubsystems

import dev.nextftc.core.commands.Command
import dev.nextftc.core.subsystems.Subsystem
import dev.nextftc.hardware.impl.ServoEx
import dev.nextftc.hardware.positionable.SetPosition

object KickerSubsystem: Subsystem {
    val kickerServo: ServoEx = ServoEx("k",-0.1)

    val engageKicker: Command = SetPosition(kickerServo, 0.25).requires(this)
    val disengageKicker: Command = SetPosition(kickerServo, 0.0).requires(this)

    override fun initialize() {
        kickerServo.position = 0.0
    }
}